### PR TITLE
solveset : solving equation for non-symbol.

### PR DESCRIPTION
**Pull request by Shekharrajak**
TODO
- [x] Docstring
- [x] Tests

Feature added : 

```
    >>> solveset(f(x) + f(x).diff(x), f(x))
    {-Derivative(f(x), x)}
    >>>solveset((3 - 5*x/f(x))*f(x), f(x))
    {5*x/3}
```

### CODE REVIEW

**Comment by Shekharrajak** [comment]
I was trying for list of symbols but after modifying 2-3 methods, I found that many other is needed changes. So I rolled back and just tried for one variable.

**Comment by Shekharrajak** [comment]
Ping @hargup @aktech @kshitij10496 @certik

**Comment by hargup** [comment]
@Shekharrajak I've given you a TODO list, please try adding at least one line of for all the functions you implement.

**Comment by Shekharrajak** [comment]
@hargup , I added description in this PR.

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -640,6 +640,93 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def check_noninverts(f, symbol):
+    """substitute the non invertable terms Derivative, f(x) etc.
```
Comment (hargup) [inline sympy/solvers/solveset.py]: Please be specific and specify what "etc" means, you can simply say Derivative, Function and Integral.

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -640,6 +640,93 @@ def _solveset(f, symbol, domain, _check=False):
...
+            s_new = temp % i
+            non_inverts[ig] = s_new
+    # other new non_inverts will be added here
+    f = f.subs(non_inverts)
+    symbol = symbol.subs(non_inverts)
+    return f, symbol, non_inverts
+
+
+def swap_back(result, swapped_symbol, non_inverts):
+    """Replace temporary variable(if it is there) with original terms
```
Comment (hargup) [inline sympy/solvers/solveset.py]: Original terms as in?

**Comment by hargup** [comment]
Please add examples to your docstring.

**Comment by smichr** [comment]
Regarding functions of more than one variable:  perhaps see `solve`. Do  you mean that the modifications you have made don't allow for the following?

```python
>>> solve(f(x,y)+2,f(x,y))
[−2] 
```

**Comment by smichr** [comment]
Also, please search for commas and make sure that a space appears after them.

**Comment by certik** [comment]
Thanks for the review @smichr. (When you get to it, can you please reply to my private email? Thanks!)

**Comment by Shekharrajak** [comment]
I didn't understand why it failed [here](https://travis-ci.org/sympy/sympy/jobs/131354900#L3957)

File: sympy/solvers/solveset.py [hunk 3]
```diff
@@ -640,6 +641,126 @@ def _solveset(f, symbol, domain, _check=False):
...
+            non_inverts[ig] = s_new
+    # other new non_inverts will be added here
+    if non_inverts:
+        f = f.subs(non_inverts)
+        symbol = symbol.subs(non_inverts)
+    return f, symbol, non_inverts
+
+
+def swap_back(result, swapped_symbol, non_inverts):
+    """Replace temporary variable(if it is there) that where swapped.
```
Comment (parsoyaarihant) [inline sympy/solvers/solveset.py]: i think its `were` instead of `where`

Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: Thanks.

File: sympy/solvers/solveset.py [hunk 4]
```diff
@@ -737,6 +858,18 @@ def solveset(f, symbol=None, domain=S.Complexes):
     >>> solveset(exp(x) > 1, x, R)
     (0, oo)
 
+    * If Non-symbol is passed :
```
Comment (hargup) [inline sympy/solvers/solveset.py]: I would write it as something like "We can also solve the equation for a non symbol"

File: sympy/solvers/solveset.py [hunk 5]
```diff
@@ -759,18 +892,43 @@ def solveset(f, symbol=None, domain=S.Complexes):
             return S.EmptySet
         else:
             raise NotImplementedError(filldedent('''
-                relationship between value and 0 is unknown: %s''' % b))
+               relationship between value and 0 is unknown: %s''' % b))
```
Comment (hargup) [inline sympy/solvers/solveset.py]: Please revert this change.

**Comment by hargup** [comment]
@Shekharrajak  In
```
>>> solveset(f(x) + f(x).diff(x), f(x))
{-Derivative(f(x), x)}
```
The solution is dependent on the mathematical object we are solving for. Ideally it should solve the differential equation or remain unevaluated. I see that you are trying to replicate the behavior of `solve` but I think this a bug of `solve` rather than a feature. If you have good reasons to keep this behavior please explain, otherwise I don't think we should go ahead with this PR.

**Comment by Shekharrajak** [comment]
Yes, we can do these things by simply assuming them as symbol. These lines of code are not needed actually. I am closing it.
